lib/ReactViews/Search/SearchBoxAndResults.tsx

```diff
@@ -67,85 +73,65 @@ const PresentationBox = styled(Box).attrs({
 
 export const LOCATION_SEARCH_INPUT_NAME = "LocationSearchInput";
 
-export class SearchBoxAndResultsRaw extends Component {
-  constructor(props) {
-    super(props);
-    this.locationSearchRef = createRef();
-  }
+interface SearchBoxAndResultsProps {
+  placeholder: string;
+}
 
-  componentDidMount() {
-    this.props.viewState.updateAppRef(
-      LOCATION_SEARCH_INPUT_NAME,
-      this.locationSearchRef
-    );
-    this.subscribeToProps();
-  }
+export const SearchBoxAndResults: FC<SearchBoxAndResultsProps> = observer(
+  ({ placeholder }) => {
+    const locationSearchRef = useRef(null);
 
-  componentDidUpdate() {
-    this.subscribeToProps();
-  }
+    const viewState = useViewState();
+    const theme = useTheme();
 
-  componentWillUnmount() {
-    this.unsubscribeFromProps();
-  }
+    useEffect(() => {
+      viewState.updateAppRef(LOCATION_SEARCH_INPUT_NAME, locationSearchRef);
 
-  subscribeToProps() {
-    this.unsubscribeFromProps();
-
-    // TODO(wing): why is this a reaction here and not in viewState itself?
-    // Close the search results when the Now Viewing changes (so that it's visible).
-    this._nowViewingChangeSubscription = reaction(
-      () => this.props.terria.workbench.items,
-      () => {
-        this.props.viewState.searchState.showLocationSearchResults = false;
-      }
-    );
-  }
+      // TODO(wing): why is this a reaction here and not in viewState itself?
+      // Close the search results when the Now Viewing changes (so that it's visible).
+      const disposeReaction = reaction(
+        () => viewState.terria.workbench.items,
+        () => {
+          viewState.searchState.showLocationSearchResults = false;
+        }
+      );
```

Comment:
Suggestion: Here again we can replace the `reaction` with a `useEffect` like:

```js
    useEffect(() => {
      viewState.searchState.showLocationSearchResults = false;
    }, [viewState.terria.workbench.items]);
```

Reply:
yeah we can do that here as it is a simple value so we will always have new reference